src/Microsoft.DotNet.Migration/ProjectMigrator.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Construction;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Cli;
+using System.Linq;
+using System.IO;
+
+namespace Microsoft.DotNet.Migration
+{
+    public class ProjectMigrator
+    {
+        // TODO: Migrate P2Ps
+        // TODO: Migrate AssemblyInfo
+        // TODO: Migrate PackOptions
+        // TODO: Migrate Scripts
+        // TODO: Migrate Configurations
+        // TODO: Support Mappings in IncludeContext Transformations
+        // TODO: Support Removal of default items/properties from template when appropriate
+        // TODO: Specify ordering of generated property/item groups (append at end of file in most cases)
+        // TODO: 
+
+        public int Migrate(string projectDirectory, string outputDirectory)
+        {
+            if (!Directory.Exists(outputDirectory))
+            {
+                Directory.CreateDirectory(outputDirectory);
+            }
+
+            var projectContexts = ProjectContext.CreateContextForEachTarget(projectDirectory);
+
+            if (projectContexts.Count() > 1)
+            {
+                throw new Exception("MultiTFM projects currently not supported.");
+            }
+
+            if (projectContexts.Count() == 0)
+            {
+                throw new Exception("No projects found");
+            }
+
+            var projectContext = projectContexts.First();
+
+            // Get template
+            var csproj = GetDefaultMSBuildProject();
+
+            // Apply Transformations
+            var ruleset = new DefaultMigrationRuleSet();
+            ruleset.Apply(projectContext, csproj, outputDirectory);
+
+            csproj.Save(Path.Combine(outputDirectory, projectContext.ProjectFile.Name + ".csproj"));
+
+            return 1;
```

Comment:
Does it make sense to return anything from this method?  Why not make it 'void' and throw exceptions on errors?  If anything, I'd assume we will want some sort of "Result" object being returned, so we can return multiple information:

1. Success/failure
2. Where the new project was created.
3. Any warnings/errors of things that couldn't be migrated.
4. etc.